freqtrade/optimize/hyperopt.py

```diff
@@ -392,8 +392,6 @@ def start(self) -> None:
         self.backtesting.exchange._api_async = None  # type: ignore
         # self.backtesting.exchange = None  # type: ignore
         self.backtesting.pairlists = None  # type: ignore
-        self.backtesting.strategy.dp = None  # type: ignore
-        IStrategy.dp = None  # type: ignore
```

Comment:
I'm ... not 100% certain this works if you use it with anything `-j > 1` ... 

Either way - i think with this it shouldn't be necessary to reassign dp to the strategy again and again in backtesting?

Reply:
I pushed some more changes which are.. Basically a different flavor of same 👀 it seems like assigning objects to class is not exactly pickle-friendly. It does not fail immediately, but after a round of hyperopts exception is throw that dp is unset. My theory is that first round might be running in the original process and still has access to this object while it does not carry on into subprocesses. I have no proof though. However we already had this problem. [This](https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/optimize/hyperopt.py#L76) used to be set to hyperopt class, not instance, and i got same issue while hyperopting - it would become unset.

Reply:
I guess if dataprovider is created without exchange, this should be fine.

Be sure to test it beyond 30 epochs though (or reduce the "random" epochs) - sometimes hyperopt reacts differently once it reaches that point - but i don't expect any problems with that
(note: i will test that myself too - but not today).